packages/server/src/controllers/Controller.js

```diff
@@ -111,6 +112,68 @@ export class Controller {
   }
 
   setupAction(type, name, handler, authorize) {
+    if (!isFunction(handler)) {
+      const setupHandlerFromParam = ({
+        handler,
+        action,
+        authorize,
+        parameters,
+        returns,
+        scope,
+        transacted
+      }) => {
+        if (action !== undefined) {
+          handler.verb = action.verb
+          handler.path = action.path
+        }
+
+        if (authorize !== undefined) {
+          handler.authorize = authorize
+        }
+
+        if (parameters !== undefined) {
+          const parametersHasOptions = isArray(parameters[0])
+          handler.parameters = parametersHasOptions
+            ? parameters[0] : parameters
+
+          // If validation options are provided, expose them through
+          // `handler.options.parameters`, see ControllerAction
+          if (parametersOparametersHasOptionstions) {
+            handler.options = handler.options || {}
+            handler.options.parameters = parameters[1]
+          }
+        }
+
+        if (returns !== undefined) {
+          const returnsHasOptions = isArray(returns)
+          handler.returns = returnsHasOptions
+            ? returns[0] : returns
+
+          // If validation options are provided, expose them through
+          // `handler.options.returns`, see ControllerAction
+          if (returnsHasOptions) {
+            handler.options = handler.options || {}
+            handler.options.returns = returns[1]
+          }
+        }
+
+        if (scope !== undefined) {
+          const handlerScope = handler.scope = handler.scope || []
+          if (isArray(scope)) {
+            handlerScope.push(...scope)
+          } else {
+            handlerScope.push(scope)
+          }
+        }
+
+        if (transacted === true) {
+          handler.transacted = true
+        }
```

Comment:
Could just be `handler.transacted = transacted`, to move the setting over always?